terraform/fdi_failure_sns_topic.tf

```diff
@@ -0,0 +1,47 @@
+# ------------------------------------------------------------------------------
+# Create the SNS topic that allows email to be sent when the findings-data-import
+# Lambda has a failure when processing an uploaded JSON.
+# ------------------------------------------------------------------------------
+
+resource "aws_sns_topic" "fdi_failure_alarm" {
+  name         = format("fdi-failure-alarms_%s", local.production_workspace ? "production" : terraform.workspace)
+  display_name = "findings-data-import_failure_alarm"
+}
+
+# Allow the findings data bucket (and only the findings data bucket) to publish
+# notifications to the fdi failure alarm SNS topic.
+data "aws_iam_policy_document" "fdi_failure_alarm" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "sns:Publish",
+    ]
+
+    principals {
+      type        = "Service"
+      identifiers = ["s3.amazonaws.com"]
+    }
+
+    resources = [aws_sns_topic.fdi_failure_alarm.arn]
+
+    condition {
+      test     = "ArnEquals"
+      variable = "aws:SourceArn"
+      values   = [data.aws_s3_bucket.findings_data.arn]
```

Comment:
Should we alphabetize these?

Reply:
See [here](https://github.com/cisagov/cyhy_amis/pull/646#discussion_r1161825858).